### PR TITLE
Flag question

### PR DESCRIPTION
**Pull request by MaksymilianK**

### CODE REVIEW

File: thesis-gui/src/components/answer/Answer.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import { IonItem, IonLabel, IonRadio } from "@ionic/react";
 import './Answer.css';
 
-const Answer: React.FC<{ name: string, showCorrect: boolean, showWrong: boolean }> = props => {
+const Answer: React.FC<{ name: string, showCorrect: boolean, showWrong: boolean, disabled: boolean }> = props => {
```
Comment (elacymerys) [inline thesis-gui/src/components/answer/Answer.tsx]: Could you please change this `props` to `{ name, showCorrect, showWrong, disabled }`?

File: thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
-                    { `Question ${props.questionNumber || ''}` }
-                </IonCardTitle>
-            </IonCardHeader>
+export const DefinitionQuestionCard: React.FC<{
+    question: string,
+    questionNumber: number,
+    category: CategoryType,
+    flagDisabled: boolean,
+    flagQuestion: () => void
+}> = props => (
```
Comment (elacymerys) [inline thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx]: Could you please replace `props` with `{ question, questionNumber, category, flagDisabled, flagQuestion }`?

File: thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx [hunk 3]
```diff
@@ -1,35 +1,49 @@
...
-                </IonNote>
-            </IonItem>
+export const PictureQuestionCard: React.FC<{
+    question: string,
+    questionNumber: number,
+    category: CategoryType,
+    authorName: string,
+    flagDisabled: boolean,
+    flagQuestion: () => void
+}> = props => (
```
Comment (elacymerys) [inline thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx]: Could you please replace `props` with `{ question, questionNumber, category, authorName, flagDisabled, flagQuestion }`?

File: thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx [hunk 4]
```diff
@@ -1,28 +1,42 @@
...
+                {`Category: ${!!props.category ? props.category.name : ''}`}
+            </IonCardSubtitle>
+            <IonCardTitle style={{display: 'flex', justifyContent: 'space-between'}}>
+                {`Question ${props.questionNumber || ''}`}
+                <IonButtons>
+                    <IonButton
+                        disabled={props.flagDisabled}
+                        onClick={props.flagQuestion}
+                    >
+                        <IonIcon slot="icon-only" icon={flagOutline}/>
```
Comment (elacymerys) [inline thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx]: I know I'm being picky 🫣but I think the **not outlined** icon is more visible

File: thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx [hunk 5]
```diff
@@ -1,35 +1,49 @@
...
+                {`Category: ${!!props.category ? props.category.name : ''}`}
+            </IonCardSubtitle>
+            <IonCardTitle style={{display: 'flex', justifyContent: 'space-between'}}>
+                {`Question ${props.questionNumber || ''}`}
+                <IonButtons>
+                    <IonButton
+                        disabled={props.flagDisabled}
+                        onClick={props.flagQuestion}
+                    >
+                        <IonIcon slot="icon-only" icon={flagOutline}/>
```
Comment (elacymerys) [inline thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx]: Same as for Definition Question Card

**Review by elacymerys — CHANGES_REQUESTED** [review]
I've finished review of the frontend part and left some comments
Please @Lozoko review the backend part

File: thesis-gui/src/components/answer/Answer.tsx (hunk 1 above)
Comment (MaksymilianK) [reply thesis-gui/src/components/answer/Answer.tsx]: Done

File: thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx (hunk 3 above)
Comment (MaksymilianK) [reply thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx]: Done

File: thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx (hunk 4 above)
Comment (MaksymilianK) [reply thesis-gui/src/components/question-cards/definition-question-card/DefinitionQuestionCard.tsx]: Done

File: thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx (hunk 5 above)
Comment (MaksymilianK) [reply thesis-gui/src/components/question-cards/picture-question-card/PictureQuestionCard.tsx]: Done

**Review by elacymerys — APPROVED** [review]

**Review by Lozoko — APPROVED** [review]
No time to question variables names, mechanism alone looks good

**Pull request merged**